configuration-rendering-controller/README.md

```diff
@@ -0,0 +1,481 @@
+
+In this proposal we are not going to change that structure, just the flow, so that we make it more widely accessible,
+flexible, easier to extend while also making other parts of our platform more simple.
+
+### Problems with the current practice
+
+- deployment must be done in a specific way via the Giant Swarm Flux installation and in a custom format (in practice: collections)
+- the generated resource is always an App CR and no other formats are supported
+- to make it work, the Giant Swarm Flux installation on the MCs use a forked version the
+  [kustomize-controller](https://github.com/giantswarm/kustomize-controller) and the
+  [fluxcd/pkg](https://github.com/giantswarm/fluxcd-pkg) package that enables KRM function support in the
+  underlying `kustomize` library, that is actually still alpha features. It was proposed to make it an option
+  to enable plugins, but was rejected upstream.
+
+There are some other problems with the capabilities of the configuration system as well:
+
+- we are currently storing configuration in Giant Swarm Catalog bound Config Maps as well. These will be only
+  resolved by `app-operator` as they are implicitly taken as the base layer that upon all other configuration
+  layers are merged
+- there is no support for providing different configurations for multiple instances of a given app
+  - there is no support for providing configuration for workload cluster apps
+
+## Proposed solution
+
+The proposal is to decouple the configuration system from App Platform. We want to implement a controller that
+renders all the configs present in the `shared-configs` and CCR repositories to a given namespace, without any
+dependency on how this configuration is used. Once the configuration is rendered, it is up to the user to use it
+wherever they want (i.e. reference it in an App CR, but potentially also elsewhere, like in a HelmRelease).
+
+![Flow](flow.png)
+
+Source: <https://miro.com/app/board/uXjVKUVwcLg=/>
+
+In this proposal we will show how we can solve some of our immediate problems listed above, simplify our setup
+and maintenance costs and open up the system for extension to solve our other problems.
+
+### The configuration rendering operator
+
+We can reuse `konfigure` as a library in a Kubernetes operator that reconciles a custom CR to pre-render all or a
+given subset of configuration to a destination namespace.
+
+This way, anything is open to mount the configuration config maps and secrets whether they are Giant Swarm App CRs,
+Flux HelmReleases or whatever comes next.
+
+App Platform supports cross namespace references to some extent. It is simply RBAC restricted for workload cluster
+operators. The unique `app-operator` can read CM and Secrets from all namespaces. We can implicitly deploy one instance
+of the new CRD with a scope covering all configured apps to a Giant Swarm restricted namespace, be it `giantswarm` or a new `giantswarm-configs`
```

Comment:
Is there a specific reason why the configuration CRD could not allow setting the namespace where the config should be with a default being the namespace where the configuration CR is in?